src/basic/efivars.c

```diff
@@ -30,48 +30,22 @@
 #define EFI_N_RETRIES_TOTAL 25
 #define EFI_RETRY_DELAY (50 * USEC_PER_MSEC)
 
-char* efi_variable_path(sd_id128_t vendor, const char *name) {
-        char *p;
-
-        if (asprintf(&p,
-                     "/sys/firmware/efi/efivars/%s-" SD_ID128_UUID_FORMAT_STR,
-                     name, SD_ID128_FORMAT_VAL(vendor)) < 0)
-                return NULL;
-
-        return p;
-}
-
-static char* efi_variable_cache_path(sd_id128_t vendor, const char *name) {
-        char *p;
-
-        if (asprintf(&p,
-                     "/run/systemd/efivars/%s-" SD_ID128_UUID_FORMAT_STR,
-                     name, SD_ID128_FORMAT_VAL(vendor)) < 0)
-                return NULL;
-
-        return p;
-}
-
 int efi_get_variable(
-                sd_id128_t vendor,
-                const char *name,
+                const char *variable,
                 uint32_t *ret_attribute,
                 void **ret_value,
                 size_t *ret_size) {
 
         _cleanup_close_ int fd = -1;
-        _cleanup_free_ char *p = NULL;
         _cleanup_free_ void *buf = NULL;
         struct stat st;
         usec_t begin = 0; /* Unnecessary initialization to appease gcc */
         uint32_t a;
         ssize_t n;
 
-        assert(name);
+        assert(variable);
 
-        p = efi_variable_path(vendor, name);
-        if (!p)
-                return -ENOMEM;
+        const char *p = strjoina("/sys/firmware/efi/efivars/", variable);
```

Comment:
hmm, APIs that don't validate sound less than ideal for me... i.e. now people can pass in `../../../../etc/shadow` as  variable name and get the shadow file back...

Reply:
It's not public api. In our internal code we call almost always call with a fixed string (generated through a macro). The only exception is the `Boot%04x` fields, where we insert the number at runtime. So we know exactly what inputs are passed here.